### PR TITLE
feat: webpay plus

### PR DESCRIPTION
**Pull request by victormendoza96**
- [x] add Webpay Plus flow

### CODE REVIEW

File: app/Http/Controllers/WebpayController.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                'msg' => $e->getMessage(),
+                'code' => $e->getCode()
+            );
+            return view('webpay.refund', ["resp" => $resp]);
+        }
+        return view('webpay.refund', ["resp" => $resp]);
+    }
+
+    public function status(Request $request){
+        // $req = $request->except('_token');
```
Comment (gasspper) [inline app/Http/Controllers/WebpayController.php:60]: delete this comment please

File: composer.json [hunk 2]
```diff
@@ -63,4 +74,4 @@
     },
     "minimum-stability": "stable",
     "prefer-stable": true
-}
+}
```
Comment (gasspper) [inline composer.json:77]: add new line here

File: resources/views/webpay/commit.blade.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    <div class="mb-32"> @dump($resp)</div>
+
+    <h2>Paso 2 - Petición:</h2>
+    <p class="mb-32">
+        Utilizarás el token recibido para confirmar la transacción mediante una nueva llamada a WebpayPlus.
+    </p>
+
+    <div class="mb-32">
+        {{-- @dump($respond) --}}snippet
+    </div>
```
Comment (gasspper) [inline resources/views/webpay/commit.blade.php:26]: why the comment, and the snippet text? u have el dump $resp above

File: routes/web.php [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
 Route::view('/oneclick-mall', 'home')->name('oneclick-mall');
 Route::view('/transaccion-completa', 'home')->name('transaccion-completa');
 Route::view('/patpass-comercio', 'home')->name('patpass');
+
+Route::prefix('webpay-plus')->name("webpay.")->group(function () {
+    Route::get('/create', [WebpayController::class, 'index'])->name("create");
+    Route::get('/commit', [WebpayController::class, 'commit'])->name("commit");
+    Route::post('/refund', [WebpayController::class, 'refund'])->name("refund");
+    Route::get('/status', [WebpayController::class, 'status'])->name("status");
+});
```
Comment (gasspper) [inline routes/web.php:17]: new line here

**Review by gasspper — COMMENTED** [review]
check the comments 😄

File: app/Http/Controllers/WebpayController.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+class WebpayController extends Controller
+{
+
+    public function __construct(){
+        WebpayPlus::configureForTesting();
+    }
+
+    public function index(){
+
+        $cretaTx = [
```
Comment (Matiasnickolas) [inline app/Http/Controllers/WebpayController.php:18]: maybe this should be _createTx_

File: public/css/app.css [hunk 6]
```diff
@@ -306,3 +517,15 @@ a {
...
+/* :root {
+    color-scheme: light dark;
+}
+
+@media (prefers-color-scheme: dark) {
+
+    body {
+        background-color: #1f242e;
+    }
+} */
```
Comment (Matiasnickolas) [inline public/css/app.css:531]: remove unused code

Comment (mastudillot) [inline public/css/app.css:531]: remove comments and add new line here

File: app/Http/Controllers/WebpayController.php [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+            "sessionId"=> "S-" . (string)(rand(1, 10000)),
+            'returnUrl' => url('/') . '/webpay-plus/commit',
+            'amount' => rand(1000, 2000)
+        ];
+
+        $resp = (new Transaction)->create($cretaTx['buyOrder'], $cretaTx['sessionId'], $cretaTx['amount'], $cretaTx['returnUrl']);
+        return view('webpay.create', ["request" => $cretaTx, "respond" => $resp]);
+    }
+    public function commit(Request $request){
+        //Flujo normal
```
Comment (mastudillot) [inline app/Http/Controllers/WebpayController.php:29]: consider the fourth flow, when you have in the request
token_ws, TBK_TOKEN, TBK_SESSION_ID, TBK_ORDEN_COMPRA

File: resources/views/partials/channels.blade.php [hunk 8]
```diff
@@ -1,26 +1,29 @@
...
-                <a href="http://google.com" target="_blank">
-                    <Image unoptimized src={{ asset('images/outlook.png') }} alt="outlook" width="32"
-                        height="32" />
-                </a>
-                <span class="container">Necesitas resolver algún tipo de incidencia, contáctanos a través de correo
-                    electrónico</span>
+            <div class="question-item">
+                <span class="title">Envíanos un mensaje</span>
+                <div class="inner-container">
+                    <a href="http://google.com" target="_blank">
```
Comment (mastudillot) [inline resources/views/partials/channels.blade.php:20]: use soporte@transbank.cl

**Review by mastudillot — COMMENTED** [review]
Check my comments

**Review by mastudillot — APPROVED** [review]
LGTM

**Review by tribiec — APPROVED** [review]

**Pull request merged**